### PR TITLE
Extend transformations data

### PR DESCRIPTION
**Pull request by LudoVio**
**What is this feature?**

- Extending transformations with categories.
- Add dots on all descriptions for concistencies. (most have, but some don't)
- Add images on the `public/img` folder with the name being the transformation id. (will be used to compute path at runtime)

**Why do we need this feature?**

Currently we just have a big list of tranformations. We are trying to redesign the transformation tab with filtering and some illustration to make easier for user to use transformation

This PR is part of a bigger one:
https://github.com/grafana/grafana/pull/70146

### CODE REVIEW

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts [hunk 1]
```diff
@@ -28,8 +28,27 @@ export interface TransformerRegistryItem<TOptions> extends RegistryItem {
...
+  /**
+   * Image path for the transformer
+   */
+  image?: { dark: string; light: string };
 }
 
+export type TransformerCategory =
+  | 'combine'
+  | 'calculateNewFields'
+  | 'createNewVisualization'
```
Comment (torkelo) [inline packages/grafana-data/src/transformations/standardTransformersRegistry.ts:46]: this is a very strange transformation category

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts [hunk 2]
```diff
@@ -28,8 +28,27 @@ export interface TransformerRegistryItem<TOptions> extends RegistryItem {
...
+  | 'filterAndReduce'
```
Comment (torkelo) [inline packages/grafana-data/src/transformations/standardTransformersRegistry.ts:47]: seems like two very different things to form a category :)

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts [hunk 3]
```diff
@@ -28,8 +28,27 @@ export interface TransformerRegistryItem<TOptions> extends RegistryItem {
...
+   */
+  categories?: Set<TransformerCategory>;
+
+  /**
+   * Image path for the transformer
+   */
+  image?: { dark: string; light: string };
 }
 
+export type TransformerCategory =
```
Comment (torkelo) [inline packages/grafana-data/src/transformations/standardTransformersRegistry.ts:43]: maybe an enum is better here?

**Review by torkelo — COMMENTED** [review]
Hm... embedding the images in grafana/data could be problematic 

we could have them in public/img/transformations and just have a naming pattern based on id (and theme)

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts (hunk 2 above)
Comment (LudoVio) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts]: Renamed that one to just "filter" as there were no reduced involved

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts (hunk 3 above)
Comment (LudoVio) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts:43]: I tried with an enum but then I couldnt export it and use it as a value from other files. Maybe there is somethign missing in our ts config, or maybe I am doing somethign wrong here

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts [hunk 4]
```diff
@@ -28,6 +28,26 @@ export interface TransformerRegistryItem<TOptions> extends RegistryItem {
...
+
+  /**
+   * Set of categories associated with the transformer
+   */
+  categories?: Set<TransformerCategory>;
+
+  /**
+   * Image path for the transformer
+   */
+  image?: { dark: string; light: string };
```
Comment (torkelo) [inline packages/grafana-data/src/transformations/standardTransformersRegistry.ts:40]: We can just use the id, and a naming pattern to get the image path

**Review by torkelo — COMMENTED** [review]
At, sorry maybe I mistook the images from being inside the data package.

But best thing would be inside the public folder. And use naming pattern based id and light / dark theme.

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts (hunk 1 above)
Comment (catherineymgui) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts]: Thanks for the feedback! Do you mean the name of the group is odd, or the existence of the group itself is odd?

Some suggestions for names:
- Setup for visualization
- Adapt for visualization 
- Setup new visualization

As for the group itself, these are somewhat the odd ones out. We did a grouping exercise w/ some developers and this ultimately this is what I ended up with.

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts (hunk 4 above)
Comment (LudoVio) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts:40]: theme (dark or light) is a runtime thing, but image path is a build time thing.

So we need to import both versions anyway, for the frontend build system to pick them up, optimize and create an output path with a hash in the name of the file (that we don't know of before end).

What can I do is set the "image" attribute type to be string, and assign the right image at runtime.

Comment (LudoVio) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts:40]: I can see that assets in the "public/img" folder are generally used without including them in the build process. So there is no optimization and no hash on filename, which is usefull for browser caching efficiency.

Do you advice to also does the same for the transformations illustration?

**Comment by catherineymgui** [comment]
Noticed some things along the stuff we talked about re: toggle, descriptions

- [ ] Content on the page: Could the description always be 8px from the header? (regardless of how much text there is)? And the images are always aligned to the bottom (8px)?
Screenshot 2023-06-27 at 20 30 51

- [ ] In some cases there’s no padding b/w the Beta badge and the title of the transformation
Screenshot 2023-06-27 at 20 31 27

- [ ] The transformations text + external link icon is responding weirdly on smaller screen sizes
Screenshot 2023-06-27 at 20 33 14

- [ ] Have 16px margin b/w the rows of transformations

**Comment by LudoVio** [comment]
@catherineymgui I think this is the wrong PR, this one is about grafana-data, so it has nothing viual on it

You can comment on this one:
https://github.com/grafana/grafana/pull/70146

File: packages/grafana-data/src/transformations/standardTransformersRegistry.ts (hunk 4 above)
Comment (LudoVio) [reply packages/grafana-data/src/transformations/standardTransformersRegistry.ts]: Did it the way you described

**Review by bohandley — APPROVED** [review]
LGTM! Good to see some structure and organization applied to types of transformations. Good work!

**Comment by samjewell** [comment]
@catherineymgui did you create the images / SVGs that were used in this PR? I'm wondering if there's a master-file where these images can be maintained / updated, that we could share openly and make available to community members too...
Did you use Figma, perhaps?

**Comment by samjewell** [comment]
> Is this something you are personally wanting?

Yes, that's right. For the hackathon actually. I'm not certain I'll use it this week, but it's on the cards

**Pull request merged**